tasks/skylab/StarAlign.wdl

```diff
@@ -345,100 +354,93 @@ task STARsoloFastq {
     echo -e "@CO\tReference genome used: ~{reference_path}" >> header.txt
     samtools reheader header.txt Aligned.sortedByCoord.out.bam > Aligned.sortedByCoord.out.reheader.bam
 
-
     echo "UMI LEN " $UMILen
-
-    touch barcodes_sn_rna.tsv
-    touch features_sn_rna.tsv
-    touch matrix_sn_rna.mtx
-    touch CellReads_sn_rna.stats
-    touch Features_sn_rna.stats
-    touch Summary_sn_rna.csv
-    touch UMIperCellSorted_sn_rna.txt
-
+    touch barcodes_sn_rna.tsv features_sn_rna.tsv matrix_sn_rna.mtx CellReads_sn_rna.stats Features_sn_rna.stats Summary_sn_rna.csv UMIperCellSorted_sn_rna.txt
+    ls -R
+
+    ###########################################################################
+    # SAVE OUTPUT FILES
+    ###########################################################################
+    # Function to move .mtx files to /cromwell_root/
+    move_mtx_files() {
+      local directory=$1
+      echo "Processing $directory"
+      find "${directory}/raw" -maxdepth 1 -type f -name "*.mtx" -print0 | xargs -0 -I{} sh -c 'echo Moving {}; mv {} /cromwell_root/'
+    }
+
+    # Function to move and rename common files
+    move_common_files() {
+      local src_dir=$1
+      local prefix=$2
+
+      declare -A files=(
+            ["barcodes.tsv"]="barcodes.tsv"
+            ["features.tsv"]="features.tsv"
+            ["CellReads.stats"]="CellReads.stats"
+            ["Features.stats"]="Features.stats"
+            ["Summary.csv"]="Summary.csv"
+            ["UMIperCellSorted.txt"]="UMIperCellSorted.txt"
+      )
+
+      for file in "${!files[@]}"; do
+          file_path="${files[$file]}"
+          name=$(basename "$file_path")
+          base="${name%.*}"
+          extension="${name##*.}"
+          new_name="${base}${prefix}.${extension}"
```

Comment:
the "prefix" is kind of being used as suffix instead because it's coming at the end. Don't know if that's important but it made me take a second glance to make sure the files were being named with _sn_rna at the end